### PR TITLE
[SYCL][NFC] Fix aspects detection of AllocaInst when opaque pointers are enabled

### PR DESCRIPTION
**Pull request by AlexeySachkov**

### CODE REVIEW

File: llvm/lib/SYCLLowerIR/SYCLPropagateAspectsUsage.cpp [hunk 1]
```diff
@@ -191,6 +191,12 @@ const AspectsSetTy &getAspectsFromType(const Type *T,
 AspectsSetTy getAspectsUsedByInstruction(const Instruction &I,
                                          TypeToAspectsMapTy &Types) {
   const Type *ReturnType = I.getType();
+  if (auto *AI = dyn_cast<const AllocaInst>(&I)) {
+    // Return type of an alloca is a pointer and in opaque pointers world we
+    // don't know which type it points to. Therefore, explicitly checking the
+    // allocated type insted
```
Comment (unterumarmung) [inline llvm/lib/SYCLLowerIR/SYCLPropagateAspectsUsage.cpp:197]:
```suggestion
    // allocated type instead
```

**Comment by bader** [comment]
@AlexeySachkov, @intel/dpcpp-tools-reviewers, ping.

File: llvm/lib/SYCLLowerIR/SYCLPropagateAspectsUsage.cpp (hunk 1 above)
Comment (AlexeySachkov) [reply llvm/lib/SYCLLowerIR/SYCLPropagateAspectsUsage.cpp]: Fixed in 477a07e42af1254d21d474a1bca224193a90c97f

**Comment by AlexeySachkov** [comment]
I apologize for the force-push, accidentally merged with the wrong branch.

**Review by asudarsa — APPROVED** [review]
LGTM. Thanks. Is there a test in place for this?

**Comment by bader** [comment]
> LGTM. Thanks. Is there a test in place for this?

Yes. There are regression tests in `sycl/llvm/test/SYCLLowerIR/PropagateAspectsUsage`, which are failing in nightly. This patch is supposed to fix these failures.

**Pull request merged**